fhir-persistence-schema/src/test/java/com/ibm/fhir/schema/derby/DerbyFhirDatabaseTest.java

```diff
@@ -162,4 +178,137 @@ protected void checkRefSequence(DerbyAdapter adapter) throws SQLException {
+                prepareTestDataForTestTableHasDataFunction(connection);
+                
+                //validate table data
+                cmd = new TableHasData("FHIRDATA", "evidence_logical_resources");
+                assertTrue(adapter.runStatement(cmd));
+                cmd = new TableHasData("FHIRDATA", "evidencevariable_logical_resources");
+                assertTrue(adapter.runStatement(cmd));
+                connection.rollback(); // roll back the changes 
+            } catch (Throwable t) {
+                connection.rollback();
+                throw t;
+            }
+        }
+     
+    }
+
+    /**
+     * Prepare test data for test table has data function.
+     *
+     * @param connnection the connection
+     * @throws SQLException the SQL exception
+     */
+    private void prepareTestDataForTestTableHasDataFunction(Connection connection) throws SQLException {
+        int resourceTypeId = getResourceType(connection); 
+        
+        long logicalResourceId = getNextLogicalId(connection); 
+        
+        final String insertLogicalResource = "INSERT INTO logical_resources(logical_resource_id, resource_type_id, logical_id, last_updated, is_deleted, parameter_hash)"
+                + " VALUES (?,?,?,?,?,?)";
+        final Timestamp lastUpdated = Timestamp.from(this.lastUpdated);
+        try (PreparedStatement ps = connection.prepareStatement(insertLogicalResource)) {
+            PreparedStatementHelper psh = new PreparedStatementHelper(ps);
+            psh.setLong(logicalResourceId)
+            .setInt(resourceTypeId)
+            .setString(EVIDENCE_LOGICAL_ID)
+            .setTimestamp(lastUpdated)
+            .setString("N")
+            .setString(parameterHash);
+            ps.executeUpdate();
+        }
+        addLogicalData(connection, logicalResourceId, lastUpdated, "evidence_logical_resources"); 
+        addLogicalData(connection, logicalResourceId, lastUpdated, "evidencevariable_logical_resources"); 
+        
+    }
+
+    /**
+     * Adds the logical data.
+     *
+     * @param connnection the connection
+     * @param logicalResourceId the logical resource id
+     * @param lastUpdated the last updated
+     * @param tableName the table name
+     * @throws SQLException the SQL exception
+     */
+    private void addLogicalData(Connection connection, long logicalResourceId, final Timestamp lastUpdated, String tableName) throws SQLException {
+        final String insertEvidenceLogicalResource =
+          "INSERT INTO " + tableName + "(logical_resource_id, logical_id, is_deleted, last_updated, version_id)"
+          + " VALUES (?,?,?,?,?)";
+          
+          try (PreparedStatement stmt = connection.prepareStatement(insertEvidenceLogicalResource)) {
+              // bind parameters
+              stmt.setLong(1, logicalResourceId);
+              stmt.setString(2, "evidence123");
+              stmt.setString(3, "N");
+              stmt.setTimestamp(4, lastUpdated); 
```

Comment:
I know it doesn't strictly matter for this test case, but we always need to provide the timezone when calling setTimestamp to avoid nasty errors. However, in this case it is sufficient just to add a little comment saying you are ignoring timezone because it doesn't matter for the test.

Reply:
this has been fixed by using PreparedStatementHelper as suggested.